src/libs/SearchParser/searchParser.peggy

```diff
@@ -5,149 +5,144 @@
+{{
   function buildFilter(operator, left, right) {
     return { operator, left, right };
   }
+}}
+
+// per-parser initializer (code executed before every parse)
+{
+  const defaultValues = {
+    type: "expense",
+    status: "all",
+    sortBy: "date",
+    sortOrder: "desc",
+  };
 
   function applyDefaults(filters) {
     return {
       ...defaultValues,
-      filters
-    };
-  }
-
-  function applyPolicyID(filtersWithDefaults) {
-    return {
-      ...filtersWithDefaults,
-      policyID: filtersWithDefaults.policyID
+      filters,
     };
   }
 
   function updateDefaultValues(field, value) {
     defaultValues[field] = value;
   }
-
-  function isDefaultField(field) {
-    return defaultValues.hasOwnProperty(field);
-  }
-
-  function isPolicyID(field) {
-    return field === 'policyID';
-  }
 }
 
-query
-  = _ filters:filterList? _ {
-    const withDefaults = applyDefaults(filters);
-    if (defaultValues.policyID) {
-        return applyPolicyID(withDefaults);
-    }
-
-    return withDefaults;
-  }
+query = _ filters:filterList? _ { return applyDefaults(filters); }
 
 filterList
   = head:filter tail:(logicalAnd filter)* {
-      const allFilters = [head, ...tail.map(([_, filter]) => filter)].filter(filter => filter !== null);
+      const allFilters = [head, ...tail.map(([_, filter]) => filter)]
+        .filter(Boolean)
+        .filter((filter) => filter.right);
       if (!allFilters.length) {
-      	return null;
-      }
-     const keywords = allFilters.filter((filter) => filter.left === "keyword" || filter.right?.left === "keyword")
-     const nonKeywords = allFilters.filter((filter) => filter.left !== "keyword" && filter.right?.left !== "keyword")
-     if(!nonKeywords.length){
-     	return keywords.reduce((result, filter) => buildFilter("or", result, filter))
-     }
-     if(!keywords.length){
-     	return nonKeywords.reduce((result, filter) => buildFilter("and", result, filter))
-     }
-
-     return buildFilter("and", keywords.reduce((result, filter) => buildFilter("or", result, filter)), nonKeywords.reduce((result, filter) => buildFilter("and", result, filter)))
-
-      return allFilters.reduce((result, filter) => buildFilter("and", result, filter));
-    }
-
-filter
-  = _ field:key? _ op:operator? _ value:identifier {
-      if (isDefaultField(field)) {
-        updateDefaultValues(field, value.trim());
         return null;
       }
 
-      if (isPolicyID(field)) {
-        updateDefaultValues(field, value.trim());
-        return null;
+      const keywords = allFilters.filter(
+        (filter) =>
+          filter.left === "keyword" || filter.right?.left === "keyword"
+      );
+      const nonKeywords = allFilters.filter(
+        (filter) =>
+          filter.left !== "keyword" && filter.right?.left !== "keyword"
+      );
+
+      const keywordFilter = buildFilter(
+        "eq",
+        "keyword",
+        keywords.map((filter) => filter.right).flat()
```

Comment:
This line introduces https://github.com/Expensify/App/issues/52923

More details: https://github.com/Expensify/App/issues/52923#issuecomment-2492749621